hack/unit-tests.sh

```diff
@@ -20,8 +20,17 @@ set -o pipefail
 
 REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
 
+OPENSHIFT_CI=${OPENSHIFT_CI:-""}
+ARTIFACT_DIR=${ARTIFACT_DIR:-""}
+
+
 cd $REPO_ROOT && \
 	source ./hack/fetch_ext_bins.sh && \
 	fetch_tools && \
 	setup_envs && \
-	make unit
+if [ "$OPENSHIFT_CI" == "true" ] && [ -n "$ARTIFACT_DIR" ]; then # detect ci environment there
+  go get -u github.com/jstemmer/go-junit-report && \
+	make unit-ci
+else
+  make unit
+fi
```

Comment:
You could probably handle all of the go-junit stuff within this file right and not need the extra make target? 
Does the following work? Could make it so we do this if and only if `ARTIFACT_DIR` is set
```
make unit | go-junit-report > $(ARTIFACT_DIR)/junit_machine_api_operator.xml
```

Reply:
>You could probably handle all of the go-junit stuff within this file right and not need the extra make target?

Reply:
>Does the following work?

Reply:
Hmm not sure 🤔 I don't really see a need to move the make unit target into this file but maybe for the case where we are extracting the JUnit that would be fine to move into this file?